group_project_v2/mixins.py

```diff
@@ -125,7 +126,201 @@ def real_user_id(self, anonymous_student_id):
         return self._known_real_user_ids[anonymous_student_id]
 
 
-class WorkgroupAwareXBlockMixin(UserAwareXBlockMixin, CourseAwareXBlockMixin):
+class SettingsMixin(object):
+
+    def _get_setting(self, setting, default):
+        result = default
+        settings_service = self.runtime.service(self, "settings")
+        if settings_service:
+            xblock_settings = settings_service.get_settings_bucket(self)
+            if xblock_settings and setting in xblock_settings:
+                result = xblock_settings[setting]
+        return result
+
+
+class AuthXBlockMixin(SettingsMixin, ProjectAPIXBlockMixin):
+
+    _DEFAULT_TA_ROLE = ('assistant',)
+
+    _ACCESS_DASHBOARD_ROLE_GROUPS_KEY = "access_dashboard_groups"
+    _ACCESS_DASHBOARD_FOR_ALL_ORGS_GROUPS_KEY = "access_dashboard_for_all_orgs_groups"
+    _TA_ROLES_KEY = 'ta_roles'
+
+    @property
+    def see_dashboard_role_perms(self):
+        """
+        :return: Returns a list of group names, user needs to be a member of
+                 any group from this list to access dashboard.
+                 Additionally he will see this dashboard
+                 filtered so he only sees students from organizations he
+                 belongs to.
+        :rtype: Iterable[str]
+        """
+        return self._get_setting(self._ACCESS_DASHBOARD_ROLE_GROUPS_KEY, [])
+
+    @property
+    def see_dashboard_for_all_orgs_perms(self):
+        """
+        :return: Returns a list of group names, user needs to be a member of
+                 any group from this list to access dashboard.
+                 Members of these group will see dashboard with users
+                 from all organizations.
+        :rtype: Iterable[str]
+        """
+        return self._get_setting(self._ACCESS_DASHBOARD_FOR_ALL_ORGS_GROUPS_KEY, [])
+
+    @property
+    def ta_roles(self):
+        """
+
+        .. note:
+
+            This returns different thing than self.see_dashboard_for_all_orgs
+            this returns a **course** role, and rest contains a user group.
+
+        :return: Returns a list of group names. Normally to access a group work
+                 view user will need to belong to a team. If user is a member
+                 of this any group from the list he will be able to access group
+                 work for role group work team.
+        :rtype: Iterable[str]
+        """
+        return self._get_setting(self._TA_ROLES_KEY, self._DEFAULT_TA_ROLE)
+
+    def can_access_dashboard(self, user_id):
+        """
+        :param user_id:
+        :return: True if user can access dashboard.
+        :rtype: bool
+        """
+        user_groups = self._user_groups(user_id)
+        return bool(user_groups & self._access_dashboard_roles)
+
+    @staticmethod
+    def check_dashboard_access_for_current_user(func):
+
+        @functools.wraps(func)
+        def wrapper(self, *args, **kwargs):
```

Comment:
@jbzdak nit: while `wrapper` is commonly used in decorators, I find it somewhat a bad name - if you've got a stacktrace, you'll see `wrapper` as function name. It will still contain line number, so eventually you'll be able to figure out what went wrong, but it might make sense to give a better name from the start so that traceback contains more context.